src/OrchardCore.Modules/Orchard.OpenId/Services/OpenIdCorsConfiguration.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Linq;
+using Microsoft.AspNetCore.Cors.Infrastructure;
+using Microsoft.AspNetCore.Modules;
+using Microsoft.Extensions.Options;
+
+namespace Orchard.OpenId.Services
+{
+    [RequireFeatures("Orchard.Cors.Mvc")]
+    public class OpenIdCorsConfiguration : IConfigureOptions<CorsOptions>
+    {
+        private readonly IOpenIdService _openIdService;
+
+        public OpenIdCorsConfiguration(IOpenIdService openIdService)
+        {
+            _openIdService = openIdService;
+        }
+
+        public void Configure(CorsOptions options)
+        {
+            var openIdSettings = _openIdService.GetOpenIdSettingsAsync().GetAwaiter().GetResult();
+            if (openIdSettings.Audiences != null && openIdSettings.Audiences.Any())
+                options.AddPolicy(Constants.OpenIdConnectPolicy, builder => builder
+                    .WithOrigins(openIdSettings.Audiences.ToArray())
+                    .AllowAnyHeader()
+                    .AllowAnyMethod()
+                    .AllowCredentials()
```

Comment:
Hmm, I didn't think about flows. 
If Admin set some external Audience, He assuming that External host can connect to API. However, I missed one case, where Audience will be the same as current host

>If you are providing tokens for accessing resources on this site Audiences should be equal to Authority url

Probably we should just check if Audience equals to current Host then ignore it?